test/integration/driver.go

```diff
@@ -103,6 +103,10 @@ func main() {
 		check((&reachability{}).run())
 		check(testRouting())
 	}
+
+	glog.Info("Cleaning up ingress secret.")
```

Comment:
Perhaps move into teardown() so ingress secret is cleaned up if check() calls fail?